### ISSUE TITLE
MSS Icon missing from startmenu after conda install

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#115]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/115) by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

The conda is installed as a "script" and can be called from the commandline. Additionally, an icon in the startmenu in the Anaconda folder would be preferabl, similar to Spyder or IPython in Windows (I assume that similar stuff exists under Linux?).

Documentation states (https://conda.io/docs/building/recipe.html) :

```
#!rst

Adding Windows start menu items

When a package is installed it can add a shortcut to the Windows start menu. Conda and conda build handle this with the package menuinst, which only supports Windows at this time. Instructions for using menuinst are available on the menuinst wiki.

The easiest way to ensure a package made with conda constructor does not install any menu shortcuts is by removing menuinst from the list of conda packages included. To do this, add the following to the constuct.yaml file:

exclude:
  - menuinst


```

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

This is related to #100

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

Without an Icon, no Icon in Startmenu, clear.
Spyder does provide an Icon in Anaconda. They mayto do this with a post-install script
https://github.com/spyder-ide/spyder/blob/master/scripts/spyder_win_post_install.py
However, their info directory also contains a menu-windows.json and the conda package "index.json" file looks quite different from the one by MSS  with an "icon" entry and "app" related stuff.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

There are several related problems 

http://stackoverflow.com/questions/12432637/pyqt4-set-windows-taskbar-icon

I give this currently not much priority. But if one works on this it had to become solved in a way  that it works on all major platforms

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

A first step would be to include MSS in the Anaconda Navigator, which didn't work on my machine. I think that this requires an "app" section (including an icon) in the meta.yaml...?

Secondly I do not think that we need an all or nothing approach. The simplest way would be to copy the approach of a conda package that does offer an icon like Spyder or IPython, if that isn't too complex.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Before implementing we need first a logo which can be used for the whole project for a long time.

We should start to discuss what the logo should present on #100. There are several designer available who could help with ideas based on a list.

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

some info

http://doc.qt.io/qt-4.8/appicon.html

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Adding the image to the mslib/msui/ directory and that small simple line works with gnome. image is then shown in navigation.

```
#!text

![diff --git a/mslib/msui/mss_pyui.py b/mslib/msui/mss_pyui.py
index 96454cf..b93efbe 100644
--- a/mslib/msui/mss_pyui.py
+++ b/mslib/msui/mss_pyui.py
@@ -190,6 +190,8 @@ class MSSMainWindow(QtWidgets.QMainWindow, ui.Ui_MSSMainWindow):
     def __init__(self, *args):
         super(MSSMainWindow, self).__init__(*args)
         self.setupUi(self)
+        self.setWindowIcon(QtGui.QIcon('mss.png'))
 
         # Reference to the flight track that is currently displayed in the
         # views.

]
```
![mss.png](https://bitbucket.org/repo/MBp6Xa/images/3030878777-mss.png)

![Bildschirmfoto von »2017-02-18 14-15-34«.png](https://bitbucket.org/repo/MBp6Xa/images/265419680-Bildschirmfoto%20von%20%C2%BB2017-02-18%2014-15-34%C2%AB.png)

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

On ubuntu / gnome a file mss.desktop created on the Desktop will show up as launcher

```
#!text

[Desktop Entry]
Name=mss
Exec=/home/mss/miniconda2/envs/mssdev/bin/mss
Terminal=true
Icon=/home/mss/miniconda2/envs/mssdev/mslib/msui/mss-logo.png
Type=Application
```

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

From the pycon.de ML

https://mail.python.org/pipermail/python-de/2017q1/004674.html

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Für Linux

d.h. in unserem Fall muss die dahin

~/.local/share/applications

dann geht auch "im starter behalten"
 ￼
und das icon dahin
./local/icons/hicolor/64x64/apps/mss........png

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

For 1.4 we will implement this as experimental feature. There are variuos desktops available, which have to be checked or for which we have to improve.
The gui will get a parameter to setup a entry into the menue

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

currently for linux solved

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

looks I won't be able to call python setup.py py2app -A
distutils.errors.DistutilsError: Setup script exited with error: This distribution is only supported on MacOSX

https://py2app.readthedocs.io/en/latest/tutorial.html
https://py2app.readthedocs.io/en/latest/tutorial.html#running-your-application

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

This is partly differently implemented for Linux and Windows in 1.4.0 
not  for OSX

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Commits: 

 * for linux: fe56821
 * for windows: 418bb99

no solution yet for OSX
